frontend/src/components/application.js

```diff
@@ -3,10 +3,133 @@
  * @requires react-router-dom
  * @author Erik Hanstad
  */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import { Button, Form, Card } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { getCompetence, sendApplication } from '../redux/actions/applyActions';
 
 function Application() {
-  return <div>Application</div>;
+  const [yoe, setYoe] = useState('');
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  const [availability, setAvailability] = useState([]);
+  const [competences, setCompetence] = useState([]);
+  const apply = useSelector((state) => state.apply);
+  const user = useSelector((state) => state.auth.user);
+  const dispatch = useDispatch();
+
+  /*
+   * changes the state for year of experience
+   */
+  const sendApply = () => {
+    const { id } = user;
+    dispatch(sendApplication({ competences, availability, id }));
+  };
+
+  /*
+   * adds a availability state
+   */
+  const addAvailability = (newAvail) => setAvailability((state) => [...state, newAvail]);
+
+  /*
+   * formats a date object to form yyyy-mm-dd
+   */
+  const formatDate = (date) => {
+    const d = new Date(date);
+    let month = d.getMonth() + 1;
+    let day = d.getDate();
+    const year = d.getFullYear();
+
+    if (month.toString().length < 2) {
+      month = `0${month}`;
+    }
+    if (day.toString().length < 2) {
+      day = `0${day}`;
+    }
+
+    return [year, month, day].join('-');
+  };
+
+  const addAvail = () => {
+    const sDate = formatDate(startDate);
+    const eDate = formatDate(endDate);
+    addAvailability({ startDate: sDate, endDate: eDate });
+    alert('Time slot added');
+  };
+
+  /*
+   * adds a competence state
+   */
+  const addCompentence = (newComp) => setCompetence((state) => [...state, newComp]);
+
+  const addCom = () => {
+    const comp = document.getElementById('compSelect').value;
+    addCompentence({ competence: comp, yoe });
+    alert('competence added');
```

Comment:
från use case:

2. The system records the data and presents the specified areas of exper-
tise.

så vi måste visa alla competences som har valts  varje gång man lägger till en, samma sak med availability.